doc/src/manual/methods.md

```diff
@@ -515,6 +515,225 @@ julia> wait(schedule(t, 1))
 "definition for Int"
 ```
 
+## Design Patterns with Parametric Methods
+
+
+While complex dispatch logic is not required for performance or usability,
+sometimes it can be the best way to express some algorithm.
+Here are a few common design patterns that come up sometimes when using dispatch in this way.
+
+### Extracting the type parameter from a super-type
+
+
+Here is the correct code template for returning the element-type `T`
+of any arbitrary subtype of `AbstractArray`:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{A}) where A<:AbstractArray{T} where T = T
+```
+using, so-called, triangular dispatch.
+
+Another way, which used to be only correct way before the advent of
+triangular dispatch, is:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{AbstractArray}) = Any
+eltype{T}(::Type{AbstractArray{T}}) = T
+eltype{T, N}(::Type{AbstractArray{T, N}}) = T
+eltype{A<:AbstractArray}(::Type{A}) = eltype(supertype(A))
+```
+
+One common mistake is to try and get the element-type by using introspection::
+
+```julia
+eltype_wrong{A<:AbstractArray}(::Type{A}) = A.parameters[1]
+```
+
+However, it is not hard to construct cases where this will fail::
+
+```julia
+type BitVector <: AbstractArray{Bool, 1}
+```
+
+Here we have created a type `BitVector` which has no parameters,
+but where the element-type is still fully specified, with `T` equal to `Bool`!
+
+
+### Building a similar type with a different type parameter
+
+When building generic code, there is often a need for constructing a similar
+object with some change made to the layout of the type.
+For instance, you might have some sort of abstract array with an arbitrary element type
+and want to write the computation on it with a specific element type.
+
+Another implication of the common mistake discussed above mentioned above,
```

Comment:
either "discussed above" or "mentioned above," one of them is redundant (and I don't think there should be a comma here)